docs/learn/learn-comparisons.md

```diff
@@ -1,63 +1,189 @@
+of transactions or state changes is established between the rollup layer and the main blockchain
+(L1).
+
+A non-interactive method relies on cryptographic proofs to validate transactions or state changes
+without requiring dispute resolution or interactive challenges. A prominent example are
+zero-knowledge proofs such as a zk-SNARK or zk-STARK is generated off-chain, proving that the
+transactions in a batch satisfy the rules of the protocol. This proofs are submitted to the L1 chain
+and verified without needing further interaction.
+
+An interactive method involves a back-and-forth process between parties to validate transactions. A
+prominent example is fraud proofs, as used in optimistic rollups. If a dispute arises, a challenge
+mechanism is triggered, where a "prover" submits evidence (fraud proof) to demonstrate that a
+transaction is invalid. This can require multiple steps, with both the challenger and defender
+submitting data to the L1 chain.
+
+Here below, the key differences between interactive and non-interactive methods.
+
+| **Aspect**         | **Interactive**                                 | **Non-Interactive**                                                     |
+| ------------------ | ----------------------------------------------- | ----------------------------------------------------------------------- |
+| **Mechanism**      | Back-and-forth dispute resolution.              | One-time proof submission and verification.                             |
+| **Latency**        | Long finality due to challenge periods.         | Near-instant finality.                                                  |
+| **Security Basis** | Economic incentives and monitoring.             | Cryptographic guarantees.                                               |
+| **Complexity**     | Simpler to implement but requires monitoring.   | Computationally intensive and harder to build.                          |
+| **Use Cases**      | Broad compatibility with existing applications. | Best for high-security applications where instant finality is critical. |
+
+Both approaches serve different needs and trade-offs, depending on the balance between scalability,
+security, latency, and flexibility required.
+
+### Zero-knowledge Rollups
+
+Zero-knowledge rollups (often called ZK rollups) are a _non-interactive_ method that utilizes
+zero-knowledge proofs to compute the validity of a particular set of state changes. Whereas
+optimistic rollups relied on fraud proofs, ZK rollups rely on cryptographic validation in the form
+of ZK proofs.
+
+Zero-knowledge rollups are significantly faster in finalization, as the cryptographic validity proof
+handles the nuance of ensuring a rollup is valid. However, the ZK rollups often suffer from
+performance due to their complexity and difficult implementation into resource-constrained
+environments. Because Turing completeness is also challenging to achieve due to this computational
```

Comment:
```suggestion
handles the nuance of ensuring a rollup is valid. However, the ZK rollups often suffer from a
performance perspective due to the complexity involved in proof generation and difficult implementation into resource-constrained environments. Because Turing completeness is also challenging to achieve due to this computational
```